src/libsync/owncloudpropagator.cpp

```diff
@@ -739,6 +739,15 @@ void PropagatorCompositeJob::slotSubJobAbortFinished()
     }
 }
 
+void PropagatorCompositeJob::appendJob(PropagatorJob *job)
+{
+    // Some jobs need to know about their parent
+    if (auto download = qobject_cast<PropagateDownloadFile *>(job))
+        download->setCompositeParent(this);
```

Comment:
Maybe this could be added in PropagatorJob itself so all the jobs would know about their parrent

Reply:
Sure. I didn't want to needlessly store the parent in all jobs. But since we don't create all job instances upfront anymore that may not be an issue. Will do.